### PR TITLE
Updated GitHub Actions documentation

### PR DESCRIPTION
**Pull request by sudomateo**
Updated the GitHub Actions documentation to refer to the new `hashicorp/setup-terraform` GitHub action located [here](https://github.com/hashicorp/setup-terraform).

The old `hashicorp/terraform-github-actions` GitHub action located [here](https://github.com/hashicorp/terraform-github-actions) action will be marked as read-only and no longer be maintained as of next week.

### CODE REVIEW

File: content/redirects.txt [hunk 1]
```diff
@@ -422,17 +422,8 @@
...
-/docs/github-actions/actions/index.html                             /docs/github-actions/configuration/index.html
-/docs/github-actions/actions/fmt.html                               /docs/github-actions/configuration/fmt.html
-/docs/github-actions/actions/init.html                              /docs/github-actions/configuration/init.html
-/docs/github-actions/actions/validate.html                          /docs/github-actions/configuration/validate.html
-/docs/github-actions/actions/plan.html                              /docs/github-actions/configuration/plan.html
-/docs/github-actions/actions/apply.html                             /docs/github-actions/configuration/apply.html
-/docs/github-actions/directories.html                               /docs/github-actions/common-tasks/working-directory.html
-/docs/github-actions/remote-backend.html                            /docs/github-actions/common-tasks/backends.html
-/docs/github-actions/terraform-versions.html                        /docs/github-actions/common-tasks/terraform-versions.html
-/docs/github-actions/workspaces.html                                /docs/github-actions/common-tasks/workspaces.html
```
[the remark this answers is not in the document]
Comment (nfagerlund) [reply content/redirects.txt:435]: Ok, yeah: We need to preserve all the existing redirects, and add new redirects for the pages this PR deletes.

[the remark this answers is not in the document]
Comment (sudomateo) [reply content/redirects.txt:435]: Added back the redirects and redirected them to the new index.

File: content/source/docs/github-actions/index.html.md [hunk 2]
```diff
@@ -1,28 +1,10 @@
...
-page_title: "Home - Terraform GitHub Actions"
-description: |-
-  Terraform GitHub Actions allow executing Terraform commands via GitHub Actions.
+page_title: "Teraform GitHub Actions"
 ---
 
 # Terraform GitHub Actions
 
-This is the documentation for Terraform GitHub Actions.
+[GitHub Actions](https://help.github.com/actions) allow you to execute one or more workflows in response to GitHub events such as a pushing a commit or opening a pull request. Through GitHub Actions, a user can automate common Terraform workflows such as installing Terraform CLI and executing various `terraform` commands. HashiCorp maintains the following GitHub Actions to help automate Terraform workflows.
```
[the remark this answers is not in the document]
Comment (sudomateo) [reply content/source/docs/github-actions/index.html.md]: Added a note addressing your "BUT" point. Let me know what you think.

File: content/source/docs/github-actions/setup-terraform.html.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+page_title: "hashicorp/setup-terraform"
+---
+
+# hashicorp/setup-terraform
+
+The `hashicorp/setup-terraform` action is a JavaScript action that sets up Terraform CLI in your GitHub Actions workflow by:
+
+* Downloading a specific version of Terraform CLI and adding it to the `PATH`.
+* Configuring the [Terraform CLI configuration file](/docs/commands/cli-config.html) with a Terraform Cloud/Enterprise hostname and API token.
+* Optionally installing a wrapper script to wrap subsequent calls of the `terraform` binary and expose its STDOUT, STDERR, and exit code as outputs named `stdout`, `stderr`, and `exitcode` respectively.
```
[the remark this answers is not in the document]
Comment (sudomateo) [reply content/source/docs/github-actions/setup-terraform.html.md]: Added this!

File: content/source/docs/github-actions/setup-terraform.html.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+The source code and reference documentation for the `hashicorp/setup-terraform` action can be found [here](https://github.com/hashicorp/setup-terraform).
+
+Let's see how to use this action in a workflow!
```
[the remark this answers is not in the document]
Comment (sudomateo) [reply content/source/docs/github-actions/setup-terraform.html.md]: I agree regarding the Learn guide! We're going to move forward with a stripped down version of this content for the website and retroactively come back and add the Learn guide.

File: content/source/docs/github-actions/setup-terraform.html.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+---
+
+# hashicorp/setup-terraform
+
+The `hashicorp/setup-terraform` action is a JavaScript action that sets up Terraform CLI in your GitHub Actions workflow by:
+
+* Downloading a specific version of Terraform CLI and adding it to the `PATH`.
+* Configuring the [Terraform CLI configuration file](/docs/commands/cli-config.html) with a Terraform Cloud/Enterprise hostname and API token.
+* Optionally installing a wrapper script to wrap subsequent calls of the `terraform` binary and expose its STDOUT, STDERR, and exit code as outputs named `stdout`, `stderr`, and `exitcode` respectively.
+
```
[the remark this answers is not in the document]
Comment (sudomateo) [reply content/source/docs/github-actions/setup-terraform.html.md:15]: Done!

File: content/source/docs/github-actions/index.html.md [hunk 6]
```diff
@@ -1,28 +1,14 @@
...
+[GitHub Actions](https://help.github.com/actions) allow you to execute one or more workflows in response to GitHub events such as a pushing a commit or opening a pull request. Through GitHub Actions, a user can automate common Terraform workflows such as installing Terraform CLI and executing various `terraform` commands.
 
-[GitHub Actions](https://developer.github.com/actions) allow you to run commands in response to GitHub events such as pushing a new commit. Terraform GitHub Actions are designed to execute Terraform commands to help you review and validate Terraform changes. When Terraform GitHub Actions are executed on a pull request, a comment may be posted on the pull request conversation detailing the outcome of the GitHub Actions step.
+-> **Note** The former [hashicorp/terraform-github-actions](https://github.com/hashicorp/terraform-github-actions/) GitHub action is no longer maintained. It has been superseded by the `hashicorp/setup-terraform` GitHub action listed below. Although no functionality has been lost, workflow files will need to be reformatted to in order to use the `hashicorp/setup-terraform` GitHub action.
```
Comment (sudomateo) [inline content/source/docs/github-actions/index.html.md:10]: Here is where I added the note for users of the old actions.

**Comment by sudomateo** [comment]
@nfagerlund I addressed all the points we chatted about. Here's what I did.

- Added back the redirects and redirected them to the new index.
- Added a note for users of the older actions.
- Adjusted the verbiage slightly in a few places.

Let me know what you think!

File: content/source/docs/github-actions/index.html.md (hunk 6 above)
Comment (nfagerlund) [inline content/source/docs/github-actions/index.html.md:10]: 👍🏼 Perfect.

**Pull request merged**